src/plugins/saved_objects_management/public/management_section/object_view/saved_object_view.tsx

```diff
@@ -85,60 +85,45 @@ export class SavedObjectEdition extends Component<
       });
   }
 
+  canViewInApp = (capabilities: Capabilities, obj?: SavedObjectWithMetadata<any>) => {
+    return obj && obj.meta.inAppUrl
+      ? get(capabilities, obj?.meta.inAppUrl?.uiCapabilitiesPath, false) &&
+          Boolean(obj?.meta.inAppUrl?.path)
+      : false;
+  };
+
   render() {
-    const {
-      capabilities,
-      notFoundType,
-      serviceRegistry,
-      http,
-      serviceName,
-      savedObjectsClient,
-    } = this.props;
-    const { type } = this.state;
+    const { capabilities, notFoundType, http, uiSettings, docLinks } = this.props;
     const { object } = this.state;
-    const { edit: canEdit, delete: canDelete } = capabilities.savedObjectsManagement as Record<
-      string,
-      boolean
-    >;
-    const canView = canViewInApp(capabilities, type) && Boolean(object?.meta.inAppUrl?.path);
-    const service = serviceRegistry.get(serviceName)!.service;
-
+    const { delete: canDelete } = capabilities.savedObjectsManagement as Record<string, boolean>;
+    const canView = this.canViewInApp(capabilities, object);
     return (
-      <div data-test-subj="savedObjectsEdit">
-        <Header
-          canEdit={canEdit}
-          canDelete={canDelete && !object?.meta.hiddenType}
-          canViewInApp={canView}
-          type={type}
-          onDeleteClick={() => this.delete()}
-          viewUrl={http.basePath.prepend(object?.meta.inAppUrl?.path || '')}
-        />
-        <EuiSpacer size="l" />
-        {notFoundType && (
-          <>
-            <EuiSpacer size="s" />
-            <NotFoundErrors type={notFoundType} />
-          </>
-        )}
-        {canEdit && (
-          <>
-            <EuiSpacer size="s" />
-            <Intro />
-          </>
-        )}
-        {object && (
-          <>
-            <EuiSpacer size="m" />
-            <Form
-              object={object}
-              savedObjectsClient={savedObjectsClient}
-              service={service}
-              editionEnabled={canEdit}
-              onSave={this.saveChanges}
+      <KibanaContextProvider services={{ uiSettings }}>
```

Comment:
I'm not strongly opposed to use context providers here, but that's not really a common practice in the SOM components codebase. Can't we just pass `uiSettings` down to the component(s) that need it?

Reply:
The `CodeEditor` from `kibana_react` expects `uiSettings` on context. I searched through the code for how other plugins pass the context down and they use `KibanaContextProvider`. Is there another way that's more consistent with how we do things in the SOM components?

Reply:
No, I wasn't aware our code editor component required this context. It's fine then.